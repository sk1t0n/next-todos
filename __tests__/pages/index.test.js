import { render, screen } from '@testing-library/react';
import Home from '../../pages/index'

describe('Home', () => {
  it('check the snapshot', () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render the heading', () => {
    render(<Home />);
    expect(screen.getByText(/todo list/i)).toBeInTheDocument();
  });
});
