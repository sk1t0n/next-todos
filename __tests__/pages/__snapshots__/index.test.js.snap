// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home check the snapshot 1`] = `
<DocumentFragment>
  <div
    class="ant-row"
  >
    <div
      class="ant-col ant-col-xs-22 ant-col-xs-offset-1 ant-col-sm-20 ant-col-sm-offset-2 ant-col-md-18 ant-col-md-offset-3 ant-col-lg-12 ant-col-lg-offset-6"
    >
      <div
        class="ant-tabs ant-tabs-top"
      >
        <div
          class="ant-tabs-nav"
          role="tablist"
        >
          <div
            class="ant-tabs-nav-wrap"
          >
            <div
              class="ant-tabs-nav-list"
              style="transform: translate(0px, 0px);"
            >
              <div
                class="ant-tabs-tab ant-tabs-tab-active"
              >
                <div
                  aria-controls="rc-tabs-test-panel-1"
                  aria-selected="true"
                  class="ant-tabs-tab-btn"
                  id="rc-tabs-test-tab-1"
                  role="tab"
                  tabindex="0"
                >
                  Show task list
                </div>
              </div>
              <div
                class="ant-tabs-tab"
              >
                <div
                  aria-controls="rc-tabs-test-panel-2"
                  aria-selected="false"
                  class="ant-tabs-tab-btn"
                  id="rc-tabs-test-tab-2"
                  role="tab"
                  tabindex="0"
                >
                  Add a task
                </div>
              </div>
              <div
                class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
              />
            </div>
          </div>
          <div
            class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
          >
            <button
              aria-controls="rc-tabs-test-more-popup"
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-hidden="true"
              class="ant-tabs-nav-more"
              id="rc-tabs-test-more"
              style="visibility: hidden; order: 1;"
              tabindex="-1"
              type="button"
            >
              <span
                aria-label="ellipsis"
                class="anticon anticon-ellipsis"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="ellipsis"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
        <div
          class="ant-tabs-content-holder"
        >
          <div
            class="ant-tabs-content ant-tabs-content-top"
          >
            <div
              aria-hidden="false"
              aria-labelledby="rc-tabs-test-tab-1"
              class="ant-tabs-tabpane ant-tabs-tabpane-active"
              id="rc-tabs-test-panel-1"
              role="tabpanel"
              tabindex="0"
            >
              <div
                class="ant-table-wrapper"
              >
                <div
                  class="ant-spin-nested-loading"
                >
                  <div
                    class="ant-spin-container"
                  >
                    <div
                      class="ant-table"
                    >
                      <div
                        class="ant-table-container"
                      >
                        <div
                          class="ant-table-content"
                        >
                          <table
                            style="table-layout: auto;"
                          >
                            <colgroup />
                            <thead
                              class="ant-table-thead"
                            >
                              <tr>
                                <th
                                  class="ant-table-cell ant-table-column-has-sorters"
                                >
                                  <div
                                    class="ant-table-column-sorters"
                                  >
                                    <span
                                      class="ant-table-column-title"
                                    >
                                      Text
                                    </span>
                                    <span
                                      class="ant-table-column-sorter ant-table-column-sorter-full"
                                    >
                                      <span
                                        class="ant-table-column-sorter-inner"
                                      >
                                        <span
                                          aria-label="caret-up"
                                          class="anticon anticon-caret-up ant-table-column-sorter-up"
                                          role="img"
                                        >
                                          <svg
                                            aria-hidden="true"
                                            data-icon="caret-up"
                                            fill="currentColor"
                                            focusable="false"
                                            height="1em"
                                            viewBox="0 0 1024 1024"
                                            width="1em"
                                          >
                                            <path
                                              d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                            />
                                          </svg>
                                        </span>
                                        <span
                                          aria-label="caret-down"
                                          class="anticon anticon-caret-down ant-table-column-sorter-down"
                                          role="img"
                                        >
                                          <svg
                                            aria-hidden="true"
                                            data-icon="caret-down"
                                            fill="currentColor"
                                            focusable="false"
                                            height="1em"
                                            viewBox="0 0 1024 1024"
                                            width="1em"
                                          >
                                            <path
                                              d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                            />
                                          </svg>
                                        </span>
                                      </span>
                                    </span>
                                  </div>
                                </th>
                                <th
                                  class="ant-table-cell ant-table-column-has-sorters"
                                >
                                  <div
                                    class="ant-table-filter-column"
                                  >
                                    <span
                                      class="ant-table-column-title"
                                    >
                                      <div
                                        class="ant-table-column-sorters"
                                      >
                                        <span
                                          class="ant-table-column-title"
                                        >
                                          Completed
                                        </span>
                                        <span
                                          class="ant-table-column-sorter ant-table-column-sorter-full"
                                        >
                                          <span
                                            class="ant-table-column-sorter-inner"
                                          >
                                            <span
                                              aria-label="caret-up"
                                              class="anticon anticon-caret-up ant-table-column-sorter-up"
                                              role="img"
                                            >
                                              <svg
                                                aria-hidden="true"
                                                data-icon="caret-up"
                                                fill="currentColor"
                                                focusable="false"
                                                height="1em"
                                                viewBox="0 0 1024 1024"
                                                width="1em"
                                              >
                                                <path
                                                  d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                                />
                                              </svg>
                                            </span>
                                            <span
                                              aria-label="caret-down"
                                              class="anticon anticon-caret-down ant-table-column-sorter-down"
                                              role="img"
                                            >
                                              <svg
                                                aria-hidden="true"
                                                data-icon="caret-down"
                                                fill="currentColor"
                                                focusable="false"
                                                height="1em"
                                                viewBox="0 0 1024 1024"
                                                width="1em"
                                              >
                                                <path
                                                  d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                                />
                                              </svg>
                                            </span>
                                          </span>
                                        </span>
                                      </div>
                                    </span>
                                    <span
                                      class="ant-dropdown-trigger ant-table-filter-trigger"
                                      role="button"
                                      tabindex="-1"
                                    >
                                      <span
                                        aria-label="filter"
                                        class="anticon anticon-filter"
                                        role="img"
                                      >
                                        <svg
                                          aria-hidden="true"
                                          data-icon="filter"
                                          fill="currentColor"
                                          focusable="false"
                                          height="1em"
                                          viewBox="64 64 896 896"
                                          width="1em"
                                        >
                                          <path
                                            d="M349 838c0 17.7 14.2 32 31.8 32h262.4c17.6 0 31.8-14.3 31.8-32V642H349v196zm531.1-684H143.9c-24.5 0-39.8 26.7-27.5 48l221.3 376h348.8l221.3-376c12.1-21.3-3.2-48-27.7-48z"
                                          />
                                        </svg>
                                      </span>
                                    </span>
                                  </div>
                                </th>
                                <th
                                  class="ant-table-cell"
                                >
                                  Action
                                </th>
                              </tr>
                            </thead>
                            <tbody
                              class="ant-table-tbody"
                            >
                              <tr
                                class="ant-table-row ant-table-row-level-0"
                                data-row-key="1"
                              >
                                <td
                                  class="ant-table-cell"
                                >
                                  <span
                                    style="text-decoration: line-through;"
                                  >
                                    Todo 1
                                  </span>
                                </td>
                                <td
                                  class="ant-table-cell"
                                >
                                  <label
                                    class="ant-checkbox-wrapper"
                                    style="display: flex; justify-content: center;"
                                  >
                                    <span
                                      class="ant-checkbox ant-checkbox-checked"
                                    >
                                      <input
                                        checked=""
                                        class="ant-checkbox-input"
                                        type="checkbox"
                                        value=""
                                      />
                                      <span
                                        class="ant-checkbox-inner"
                                      />
                                    </span>
                                  </label>
                                </td>
                                <td
                                  class="ant-table-cell"
                                >
                                  <button
                                    class="ant-btn ant-btn-primary ant-btn-block ant-btn-dangerous"
                                    type="button"
                                  >
                                    <span>
                                      Remove
                                    </span>
                                  </button>
                                </td>
                              </tr>
                              <tr
                                class="ant-table-row ant-table-row-level-0"
                                data-row-key="2"
                              >
                                <td
                                  class="ant-table-cell"
                                >
                                  <span>
                                    Todo 2
                                  </span>
                                </td>
                                <td
                                  class="ant-table-cell"
                                >
                                  <label
                                    class="ant-checkbox-wrapper"
                                    style="display: flex; justify-content: center;"
                                  >
                                    <span
                                      class="ant-checkbox"
                                    >
                                      <input
                                        class="ant-checkbox-input"
                                        type="checkbox"
                                        value=""
                                      />
                                      <span
                                        class="ant-checkbox-inner"
                                      />
                                    </span>
                                  </label>
                                </td>
                                <td
                                  class="ant-table-cell"
                                >
                                  <button
                                    class="ant-btn ant-btn-primary ant-btn-block ant-btn-dangerous"
                                    type="button"
                                  >
                                    <span>
                                      Remove
                                    </span>
                                  </button>
                                </td>
                              </tr>
                              <tr
                                class="ant-table-row ant-table-row-level-0"
                                data-row-key="3"
                              >
                                <td
                                  class="ant-table-cell"
                                >
                                  <span
                                    style="text-decoration: line-through;"
                                  >
                                    Todo 3
                                  </span>
                                </td>
                                <td
                                  class="ant-table-cell"
                                >
                                  <label
                                    class="ant-checkbox-wrapper"
                                    style="display: flex; justify-content: center;"
                                  >
                                    <span
                                      class="ant-checkbox ant-checkbox-checked"
                                    >
                                      <input
                                        checked=""
                                        class="ant-checkbox-input"
                                        type="checkbox"
                                        value=""
                                      />
                                      <span
                                        class="ant-checkbox-inner"
                                      />
                                    </span>
                                  </label>
                                </td>
                                <td
                                  class="ant-table-cell"
                                >
                                  <button
                                    class="ant-btn ant-btn-primary ant-btn-block ant-btn-dangerous"
                                    type="button"
                                  >
                                    <span>
                                      Remove
                                    </span>
                                  </button>
                                </td>
                              </tr>
                              <tr
                                class="ant-table-row ant-table-row-level-0"
                                data-row-key="4"
                              >
                                <td
                                  class="ant-table-cell"
                                >
                                  <span
                                    style="text-decoration: line-through;"
                                  >
                                    Todo 4
                                  </span>
                                </td>
                                <td
                                  class="ant-table-cell"
                                >
                                  <label
                                    class="ant-checkbox-wrapper"
                                    style="display: flex; justify-content: center;"
                                  >
                                    <span
                                      class="ant-checkbox ant-checkbox-checked"
                                    >
                                      <input
                                        checked=""
                                        class="ant-checkbox-input"
                                        type="checkbox"
                                        value=""
                                      />
                                      <span
                                        class="ant-checkbox-inner"
                                      />
                                    </span>
                                  </label>
                                </td>
                                <td
                                  class="ant-table-cell"
                                >
                                  <button
                                    class="ant-btn ant-btn-primary ant-btn-block ant-btn-dangerous"
                                    type="button"
                                  >
                                    <span>
                                      Remove
                                    </span>
                                  </button>
                                </td>
                              </tr>
                              <tr
                                class="ant-table-row ant-table-row-level-0"
                                data-row-key="5"
                              >
                                <td
                                  class="ant-table-cell"
                                >
                                  <span>
                                    Todo 5
                                  </span>
                                </td>
                                <td
                                  class="ant-table-cell"
                                >
                                  <label
                                    class="ant-checkbox-wrapper"
                                    style="display: flex; justify-content: center;"
                                  >
                                    <span
                                      class="ant-checkbox"
                                    >
                                      <input
                                        class="ant-checkbox-input"
                                        type="checkbox"
                                        value=""
                                      />
                                      <span
                                        class="ant-checkbox-inner"
                                      />
                                    </span>
                                  </label>
                                </td>
                                <td
                                  class="ant-table-cell"
                                >
                                  <button
                                    class="ant-btn ant-btn-primary ant-btn-block ant-btn-dangerous"
                                    type="button"
                                  >
                                    <span>
                                      Remove
                                    </span>
                                  </button>
                                </td>
                              </tr>
                            </tbody>
                          </table>
                        </div>
                      </div>
                    </div>
                    <ul
                      class="ant-pagination ant-table-pagination ant-table-pagination-right"
                      unselectable="unselectable"
                    >
                      <li
                        aria-disabled="true"
                        class="ant-pagination-prev ant-pagination-disabled"
                        title="Previous Page"
                      >
                        <button
                          class="ant-pagination-item-link"
                          disabled=""
                          tabindex="-1"
                          type="button"
                        >
                          <span
                            aria-label="left"
                            class="anticon anticon-left"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="left"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                              />
                            </svg>
                          </span>
                        </button>
                      </li>
                      <li
                        class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                        tabindex="0"
                        title="1"
                      >
                        <a
                          rel="nofollow"
                        >
                          1
                        </a>
                      </li>
                      <li
                        class="ant-pagination-item ant-pagination-item-2"
                        tabindex="0"
                        title="2"
                      >
                        <a
                          rel="nofollow"
                        >
                          2
                        </a>
                      </li>
                      <li
                        class="ant-pagination-item ant-pagination-item-3"
                        tabindex="0"
                        title="3"
                      >
                        <a
                          rel="nofollow"
                        >
                          3
                        </a>
                      </li>
                      <li
                        aria-disabled="false"
                        class="ant-pagination-next"
                        tabindex="0"
                        title="Next Page"
                      >
                        <button
                          class="ant-pagination-item-link"
                          tabindex="-1"
                          type="button"
                        >
                          <span
                            aria-label="right"
                            class="anticon anticon-right"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="right"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                              />
                            </svg>
                          </span>
                        </button>
                      </li>
                      <li
                        class="ant-pagination-options"
                      >
                        <div
                          class="ant-select ant-pagination-options-size-changer ant-select-single ant-select-show-arrow"
                        >
                          <div
                            class="ant-select-selector"
                          >
                            <span
                              class="ant-select-selection-search"
                            >
                              <input
                                aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                                aria-autocomplete="list"
                                aria-controls="rc_select_TEST_OR_SSR_list"
                                aria-haspopup="listbox"
                                aria-owns="rc_select_TEST_OR_SSR_list"
                                autocomplete="off"
                                class="ant-select-selection-search-input"
                                id="rc_select_TEST_OR_SSR"
                                readonly=""
                                role="combobox"
                                style="opacity: 0;"
                                type="search"
                                unselectable="on"
                                value=""
                              />
                            </span>
                            <span
                              class="ant-select-selection-item"
                              title="5 / page"
                            >
                              5 / page
                            </span>
                          </div>
                          <span
                            aria-hidden="true"
                            class="ant-select-arrow"
                            style="user-select: none;"
                            unselectable="on"
                          >
                            <span
                              aria-label="down"
                              class="anticon anticon-down ant-select-suffix"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="down"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                />
                              </svg>
                            </span>
                          </span>
                        </div>
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              aria-labelledby="rc-tabs-test-tab-2"
              class="ant-tabs-tabpane"
              id="rc-tabs-test-panel-2"
              role="tabpanel"
              style="display: none;"
              tabindex="-1"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
